doc/develop/west/build-flash-debug.rst

```diff
@@ -311,6 +337,63 @@ For example, to build with 4 cores::
 
 The ``-o`` option is described further in the previous section.
 
+.. _west-domains-file:
+
+Domains file
+------------
+
+A domains file specifies multiple domains which can be built, flashed, and
+debugged individually using the ``--domain <domain>`` argument.
+
+The domains file is named ``domains.yaml`` and must be located in the top-level
+build folder of a multi-domain build.
```

Comment:
The unclean separation between, and lack of consistency between, the sysbuild and west build documentation could use a bit of improvement, I think.

Here are some example issues in my view:

1. We're using "multi-domain build" here in this document, but that is not a defined term in the sysbuild documentation guide, so that's confusing.
2. This is also being placed under the "examples" section; but it doesn't belong there. Did you build the docs HTML and check the section placement?
3. A definition of the contents of the domains file really don't belong in the west build tool docs, IMO -- the file is generated by sysbuild, so the docs should appear in the sysbuild documentation

Please move the parts that belong to the sysbuild documentation to the sysbuild guide, and add a separate standalone section under the west build section here that documents the sysbuild integration, referring to the sysbuild docs as necessary. The software layering is cmake -> sysbuild -> west build, so the documentation layering should match that.

Reply:
> 1. We're using "multi-domain build" here in this document, but that is not a defined term in the sysbuild documentation guide, so that's confusing.

Good point.
Let me add that to the definition overview.
But I don't think inside sysbuild itself there is much need for using this term, cause it's not until you really want to point out a specific domain of the multiple we really need to differentiate.

> 2. This is also being placed under the "examples" section, but it doesn't belong there. Did you build the docs HTML and check the section placement?

I did build it and I placed it where it made sense compared to description / examples of other build options, but I didn't pay attention to how it looked in TOC, so I didn't notice the overall examples part.
Will move it.

> 3. A definition of the contents of the domains file really don't belong in the west build tool docs, IMO -- the file is generated by sysbuild, so the docs should appear in the sysbuild documentation

But the file is only used by `west`.
If you don't use `west` and only uses CMake + ninja, then you don't care for `domains.yml`.
I see this file as similar to `runners.yml` which I consider a `west` owned file, at least the format of the file is owned by `west`, but of course generated by the Zephyr build system here:
https://github.com/zephyrproject-rtos/zephyr/blob/236484456f30576271f91b157840f049afade918/cmake/flash/CMakeLists.txt#L111-L113
`runners.yaml` doesn't seem to be documented, but if it were to be documented I would argue that it belongs with west commands like flash and debug.

You may say we don't want to document domains.yaml, because runners.yaml is not documented, and I'm fine with that and can remove this part, but if we are to describe domains.yaml file then I think it belongs as part of above mentioned west commands.

Reply:
> If you don't use west and only uses CMake + ninja, then you don't care for domains.yml.

OK, I didn't know that, and that's a fair point.

I'd say just remove the documentation for it for now so we can break it as often as we want in the early days. If it settles down and we want to make the format public, we can do it later. OK with you?

Reply:
> I'd say just remove the documentation for it for now so we can break it as often as we want in the early days.

Good plan :+1:

Reply:
> > I'd say just remove the documentation for it for now so we can break it as often as we want in the early days.
> 
> Good plan :+1:

Done

Reply:
> * We're using "multi-domain build" here in this document, but that is not a defined term in the sysbuild documentation guide, so that's confusing.
> 

Done.

> * This is also being placed under the "examples" section, but it doesn't belong there. Did you build the docs HTML and check the section placement?

Split into description and an example.
Description moved up and generally cleaned up.